src/vs/workbench/api/browser/mainThreadSecretState.ts

```diff
@@ -57,6 +57,8 @@ export class MainThreadSecretState extends Disposable implements MainThreadSecre
 			await this.credentialsService.setPassword(fullKey, key, encryptedForSet);
 			const passwordEncrypted = await this.credentialsService.getPassword(fullKey, key);
 			decrypted = passwordEncrypted && await this.encryptionService.decrypt(passwordEncrypted);
+		} finally {
+			this.credentialsService.deletePassword(fullKey, key);
```

Comment:
By doing this unconditionally in the new `finally` block aren't you **always** deleting the stored password, even one for which there isn't a problem?